plugins/synapticsmst/synapticsmst-device.c

```diff
@@ -1132,11 +1091,6 @@ synapticsmst_device_write_firmware (SynapticsMSTDevice *device,
 	else
 		chip_type = SYNAPTICSMST_CHIP_KIND_TESLA_LEAF;
 
-	if (!synapticsmst_device_check_firmware_content (device, fw, chip_type, error)){
-		g_prefix_error (error, "Invalid file content: ");
-		return FALSE;
-	}
-
```

Comment:
I think that more `FirmwareSizeMax` declarations will be needed in https://github.com/hughsie/fwupd/blob/master/plugins/synapticsmst/synapticsmst.quirk as well

Reply:
I'm not sure, `SynapticsMSTBoardID=` quirks are used on the `SynapticsMSTDevice` objects, rather than the `FuDevice` ones for instance `Guid=MST-panamera-vmm5331-259` in `dell-dock.quirk`. I think it's better to check the firmware size in the engine than just before the firmware flash.

Reply:
Well if that's the case then maybe some of this detection logic to set maximum should stay but rely upon engine for checking the bounds..  There is no indication in the quirk files for all the WD15/TB16 and system stuff that it's using a leaf device.  It relies upon automatic detection.

Reply:
I think a cleaner way to do it would be to add an extra GUID of `MST-panamera` which we could tag with attributes in the quirk files. Agree?

Reply:
Sure, that works too.  Presumably one for `MST-tesla` and `MST-leaf` too then to cover those two products.

Reply:
Ohh, i had no idea `tesla` and `leaf` were different models. How do you tell them apart?

Reply:
`synapticsmst_common_read (connection, REG_CHIP_ID,` reads the chip ID.  It's 2 bytes.
I don't know off hand what Tesla and Leaf identify as, nor if there are individual different chips in the Tesla an Leaf families.  You might have to talk to Synaptics if you want to get the hard details in here.

You'll notice current code looks just to see from a line in the sand where the delimiter is.
https://github.com/hughsie/fwupd/blob/640faed956fad10c5225b7a2abb6b64c6330b107/plugins/synapticsmst/synapticsmst-device.c#L1130

Reply:
Okay, I've just copied the same logic. Anything not using those default values can override it with a quirk.

Reply:
Cool